dcwposix/selectablemessagesocket.h

```diff
@@ -4,15 +4,13 @@
 #include "dcw/messagesocket.h"
 #include "./selecteventreactor.h"
 
-extern "C" {
-  typedef struct dcw_socket * dcw_socket_t;
-};
+using dcw_socket_t = struct dcw_socket *;
```

Comment:
It looks like this change is causing a problem for me:
```
In file included from selectablemessagesocket.cxx:3:0:
./selectablemessagesocket.h:7:7: error: expected nested-name-specifier before ‘dcw_socket_t’
 using dcw_socket_t = struct dcw_socket *;
       ^
./selectablemessagesocket.h:25:3: error: ‘dcw_socket_t’ does not name a type
   dcw_socket_t _sock;
   ^
selectablemessagesocket.cxx: In constructor ‘dcwposix::SelectableMessageSocket::SelectableMessageSocket(const char*)’:
selectablemessagesocket.cxx:36:3: error: class ‘dcwposix::SelectableMessageSocket’ does not have any field named ‘_sock’
   _sock(dcwsock_open(interfaceName)) {
   ^
selectablemessagesocket.cxx:37:7: error: ‘_sock’ was not declared in this scope
   if (_sock == NULL) {
       ^
selectablemessagesocket.cxx: In destructor ‘virtual dcwposix::SelectableMessageSocket::~SelectableMessageSocket()’:
selectablemessagesocket.cxx:46:17: error: ‘_sock’ was not declared in this scope
   dcwsock_close(_sock);
                 ^
selectablemessagesocket.cxx: In member function ‘virtual void dcwposix::SelectableMessageSocket::ReceiveMessage(dcw::MacAddress&, dcw::Message&)’:
selectablemessagesocket.cxx:53:21: error: ‘_sock’ was not declared in this scope
   rv = dcwsock_recv(_sock, buf, sizeof(buf), source.Value);
                     ^
selectablemessagesocket.cxx: In member function ‘virtual void dcwposix::SelectableMessageSocket::TransmitMessage(const dcw::MacAddress&, const dcw::Message&)’:
selectablemessagesocket.cxx:69:20: error: ‘_sock’ was not declared in this scope
   if (dcwsock_send(_sock, buf, len, dest.Value) != static_cast<int>(len)) {
                    ^
selectablemessagesocket.cxx: In member function ‘virtual void dcwposix::SelectableMessageSocket::GetSourceMacAddress(dcw::MacAddress&)’:
selectablemessagesocket.cxx:75:23: error: ‘_sock’ was not declared in this scope
   dcwsock_get_macaddr(_sock, source.Value);
                       ^
selectablemessagesocket.cxx: In member function ‘virtual int dcwposix::SelectableMessageSocket::GetSelectableFd() const’:
selectablemessagesocket.cxx:79:25: error: ‘_sock’ was not declared in this scope
   return dcwsock_get_fd(_sock);
                         ^
Makefile:433: recipe for target 'selectablemessagesocket.lo' failed
make[1]: *** [selectablemessagesocket.lo] Error 1
make[1]: Leaving directory '/var/lib/jenkins/workspace/dcwifi_github_validation/linux/dcwapd/dcwposix'
Makefile:389: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

Reply:
Fixed.